indexing-hadoop/src/main/java/io/druid/indexer/JobHelper.java

```diff
@@ -454,7 +454,25 @@ public long push() throws IOException
         .withSize(size.get())
         .withBinaryVersion(SegmentUtils.getVersionFromDir(mergedBase));
 
-    if (!renameIndexFiles(outputFS, tmpPath, finalIndexZipFilePath)) {
+    int retryAttempts = 0;
+    int retryThreshold = 5;
+    int retryGranularity = 1000;
+
+    while (retryAttempts != retryThreshold) {
```

Comment:
Note that RetryUtils does things like retry jitter io.druid.java.util.common.RetryUtils#awaitNextRetry